src/main/scala/mesosphere/marathon/core/task/termination/impl/Terminal.scala

```diff
@@ -1,18 +1,23 @@
 package mesosphere.marathon.core.task.termination.impl
 
-import mesosphere.marathon.core.event.MesosStatusUpdateEvent
+import mesosphere.marathon.core.event.InstanceChanged
+import mesosphere.marathon.core.instance.InstanceStatus
 
+// TODO(PODS): There are various similar Terminal extractors, Sets and functions – the NEED to be aligned
```

Comment:
There is already a Terminal trait and extractor, that should reflect exactly those cases - or?